packages/@expo/fingerprint/src/__tests__/Fingerprint-test.ts

```diff
@@ -372,3 +372,27 @@ describe(diffFingerprints, () => {
     ]);
   });
 });
+
+describe('function api stability', () => {
+  const Fingerprint = require('../index');
+  afterEach(() => {
+    vol.reset();
+  });
+
+  it('maintains consistent hash and function signature expected by eas-cli', async () => {
+    vol.fromJSON(require('../sourcer/__tests__/fixtures/ExpoManaged47Project.json'));
+    const fingerprintWithDebug = await Fingerprint.createFingerprintAsync('/app', {
+      platforms: ['android', 'ios'],
+      ignorePaths: ['android/**/*', 'ios/**/*'],
+      debug: true,
+    });
+    expect(fingerprintWithDebug.hash).toBe('1b77e8d7db6834f69b18e9f1cd04c03964a59312');
```

Comment:
I thing hash stability isn't necessary since it is guaranteed as long as the version locked to the sdk doesn't change. I think we should expect the signature to change with almost every change to this package.

What is more important is function signature stability (arguments and order of arguments) to the functions that are called cross-package from unversioned code. This is generally why I've preferred to do cross-package calls via a CLI (easier during code review to tell when a change is a breaking change to an API/CLI called from unversioned code). But a test that ensures a stable function signature works, too.

Reply:
> I thing hash stability isn't necessary since it is guaranteed as long as the version locked to the sdk doesn't change. I think we should expect the signature to change with almost every change to this package.

i agree this. i asked ai to help generate a test code to test the structure. maybe we can do like this
```ts
describe('test structure', () => {
  it('should keep stable output from createFingerprintAsync', async () => {
    vol.fromJSON(require('../sourcer/__tests__/fixtures/ExpoManaged47Project.json'));
    const fingerprint = await createFingerprintAsync(
      '/app',
      await normalizeOptionsAsync('/app', { debug: true })
    );

    expect(fingerprint).toEqual(
      expect.objectContaining({
        sources: expect.arrayContaining([
          expect.objectContaining({
            type: expect.stringMatching(/file|dir|contents/),
            reasons: expect.arrayContaining([expect.any(String)]),
            hash: expect.any(String),
          }),
        ]),
        hash: expect.any(String),
      })
    );

    fingerprint.sources.forEach((source) => {
      if (source.type === 'file' || source.type === 'dir') {
        expect(source).toEqual(
          expect.objectContaining({
            filePath: expect.any(String),
          })
        );
      } else if (source.type === 'contents') {
        expect(source).toEqual(
          expect.objectContaining({
            id: expect.any(String),
            contents: expect.any(String),
          })
        );
      }
    });
  });
});
```

> This is generally why I've preferred to do cross-package calls via a CLI (easier during code review to tell when a change is a breaking change to an API/CLI called from unversioned code).

i feel like we also need a way to test cli output for stable changes. no much difference between api/cli in my opinion and open to either way still

Reply:
i updated the test to check that a set of arguments in fixed order produces the output shape we expect. I used most of kudo's suggestion, except `await normalizeOptionsAsync('/app', { debug: true })` since that's not exported in the library and not available for eas-cli to call